plugins/chat/app/jobs/regular/chat/notify_watching.rb

```diff
@@ -76,7 +76,17 @@ def send_notifications(membership)
         }
 
         if membership.desktop_notifications_always? && !membership.muted?
-          ::MessageBus.publish("/chat/notification-alert/#{user.id}", payload, user_ids: [user.id])
+          send_notification =
+            !DiscoursePluginRegistry.push_notification_filters.any? do |filter|
+              !filter.call(user, payload)
```

Comment:
readability golf (avoiding `not(all?(not))` 😄)
```suggestion
            DiscoursePluginRegistry.push_notification_filters.all? do |filter|
              filter.call(user, payload)
```